src/test/txvalidation_tests.cpp

```diff
@@ -47,4 +49,51 @@ BOOST_FIXTURE_TEST_CASE(tx_mempool_reject_coinbase, TestChain100Setup)
     BOOST_CHECK(result.m_state.GetResult() == TxValidationResult::TX_CONSENSUS);
 }
 
+// Placeholder transactions that have no meaning.
+inline CTransactionRef create_placeholder_tx(size_t num_inputs, size_t num_outputs)
+{
+    CMutableTransaction mtx = CMutableTransaction();
+    mtx.vin.resize(num_inputs);
+    mtx.vout.resize(num_outputs);
+    for (size_t i{0}; i < num_inputs; ++i) {
+        mtx.vin[i].prevout.hash = InsecureRand256();
+        mtx.vin[i].prevout.n = 0;
+    }
+    for (size_t o{0}; o < num_outputs; ++o) {
+        mtx.vout[o].nValue = 1 * CENT;
+        mtx.vout[o].scriptPubKey = CScript();
+    }
+    return MakeTransactionRef(mtx);
+}
+
+BOOST_FIXTURE_TEST_CASE(package_limits, TestChain100Setup)
+{
+    // Packages can't have more than 25 transactions.
+    LOCK(cs_main);
+    Package package_too_many;
+    package_too_many.resize(MAX_PACKAGE_COUNT + 1);
```

Comment:
oh, and when I change the size to 10 here (i.e. having 10 null transactions and 10 regular ones) I get a memory access violation, so `ProcessNewPackage` seems to have a problem with malformed input. This may be related to the fuzzer crashes @adamjonas mentioned above, but I can't see any older discussion around this topic because github sucks :cry:

Reply:
agh, good catch! I meant to use `reserve()`

Reply:
Added checks for `txns.empty()` and null transactions.... hopefully this covers it.